src/components/LogOutFAB/LogOutFAB.js

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react';
+import fire from '../../fire';
+import './LogOutFAB.css';
+
+class LogOutFAB extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            loggedIn: false
+        };
+    }
+    componentWillMount() {
+        fire.auth().onAuthStateChanged((user) => {
+            if (!user) {
+                return;
+            }
+            this.setState({ loggedIn: true });
+        });
+    }
+    handleClick() {
+        fire.auth().signOut().then(function() {
+            this.setState({ loggedIn: false });
```

Comment:
`this` is undefined here, so it's erroring (silently, not sure what's catching it). Because it errors, the next line isn't getting called and the page doesn't refresh. Because the page doesn't refresh, the logged-in user's content stays on the screen (even though the login button is shown again). It's an easy fix, just use an arrow function for the `then` callback, so it automatically binds to the class context:

```
fire.auth().signOut().then(() => {
  this.setState({ loggedIn: false });
  window.location.reload(true);
}).catch(error =>  {})
```